### ISSUE TITLE
Bundler doesn't install to vendor/bundle if gem is installed globally

### ISSUE THREAD

**Issue by sled** [rubygems/rubygems#5697]
We're currently having an issue where bundler is using a globally installed gem instead of taking it from `vendor/cache` and moving it to `vendor/bundle`. This leads to a problem as soon as the ruby script requires the bundle and exits with:

```
Could not find stringio-3.0.2 in any of the sources
Run `bundle install` to install missing gems.
```

Here's a simple setup to reproduce the problem with the `stringio` gem:

### Files
**Gemfile**
```
# frozen_string_literal: true

source 'https://rubygems.org'
ruby '>= 2.7'

gem 'stringio', '3.0.2'
```
**Gemfile.lock**
```
GEM
  remote: https://rubygems.org/
  specs:
    stringio (3.0.2)

PLATFORMS
  x86_64-linux

DEPENDENCIES
  stringio (= 3.0.2)

RUBY VERSION
   ruby 2.7.6p219

BUNDLED WITH
   2.3.17
```
**test.rb**
```
#!/usr/bin/env ruby

require 'rubygems'
require 'bundler/setup'
require 'stringio'

puts StringIO.new("Hello").read
```

### Steps to reproduce the error

Install stringio globally: 
```
$>gem install stringio
Fetching stringio-3.0.2.gem
Building native extensions. This could take a while...
Successfully installed stringio-3.0.2
Parsing documentation for stringio-3.0.2
Installing ri documentation for stringio-3.0.2
Done installing documentation for stringio after 0 seconds
1 gem installed
```
Cache the bundle: 
```
$>bundle cache --all --no-install
Using bundler 2.3.17
Using stringio 3.0.2
Bundle complete! 1 Gemfile dependency, 2 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
Updating files in vendor/cache
  * stringio-3.0.2.gem
```
Install the bundle from cache to `vendor/bundle`
```
$>bundle install --local --path vendor/bundle
Using bundler 2.3.17
Using stringio 3.0.2
Updating files in vendor/cache
Bundle complete! 1 Gemfile dependency, 2 gems now installed.
Bundled gems are installed into `./vendor/bundle`
```
Run the test script:
```
$>ruby test.rb
Could not find stringio-3.0.2 in any of the sources
Run `bundle install` to install missing gems.
```

Verify the `vendor/bundle` folder is empty although the install previously stated that "Bundled gems are installed into `./vendor/bundle"
```
$>ll vendor/bundle/ruby/2.7.0/
total 8
drwxrwxr-x 2 simon simon 4096 Jul  7 18:43 ./
drwxrwxr-x 3 simon simon 4096 Jul  7 18:43 ../
```

### Behaviour if the gem is not globally installed 

Remove if installed: `gem uninstall stringio`

Cache the gems:
```
$>bundle cache --all --no-install
Fetching gem metadata from https://rubygems.org/.
Using bundler 2.3.17
Fetching stringio 3.0.2
Bundle complete! 1 Gemfile dependency, 2 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
Updating files in vendor/cache
  * stringio-3.0.2.gem
```
Install the gems:
```
$>bundle install --local --path vendor/bundle
Using bundler 2.3.17
Installing stringio 3.0.2 with native extensions
Updating files in vendor/cache
Bundle complete! 1 Gemfile dependency, 2 gems now installed.
Bundled gems are installed into `./vendor/bundle`
```
Run the test script:
```
$>ruby test.rb
Hello
```
Check the gem has been installed to `vendor/bundle`:
```
$>ll vendor/bundle/ruby/2.7.0/gems/
total 12
drwxrwxr-x 3 simon simon 4096 Jul  7 18:56 ./
drwxrwxr-x 8 simon simon 4096 Jul  7 18:56 ../
drwxr-xr-x 4 simon simon 4096 Jul  7 18:56 stringio-3.0.2/
```

**Comment by deivid-rodriguez** [comment]
Are you using the latest RubyGems? If not, can you upgrade RubyGems with `gem update --system` and try again? I tried myself and while I run into issues with older RubyGems, they disappeared after upgrading.

**Comment by sled** [comment]
Hei @deivid-rodriguez, thanks for the hint - I have the default RubyGems version installed that shipped with Ubuntu 20.04.3 LTS or Ruby 2.7.6

```
$>gem --version
3.1.6
```

I tested different RubyGems versions and it looks like it works properly from 3.2.0.rc.1 onwards.
